tests/e2e/lba_procedures/read_permission_queries.py

```diff
@@ -0,0 +1,170 @@
+        match_by_label_property_value_query,
+    ]
+
+    for grant_group in grant_groups:
+        reset_permissions(admin_cursor)
+        execute_and_fetch_all(admin_cursor, "CREATE INDEX ON :read_label;")
+        execute_and_fetch_all(admin_cursor, "CREATE INDEX ON :read_label(prop);")
+
+        [execute_and_fetch_all(admin_cursor, x) for x in grant_group]
+        for mq in match_queries:
+            results = execute_and_fetch_all(user_cursor, mq)
+            assert len(results) == 1
+
+
+def test_can_not_read_node_without_index():
+    admin_cursor = get_admin_cursor()
+    user_cursor = get_user_cursor()
+
+    grant_groups = [
+        [],
+        ["DENY READ ON LABELS :read_label TO user;"],
+        ["DENY READ ON LABELS * TO user;"],
+        [
+            "GRANT UPDATE ON LABELS :read_label TO user;",
+            "DENY READ ON LABELS :read_label TO user",
+        ],
+        [
+            "GRANT UPDATE ON LABELS * TO user;",
+            "DENY READ ON LABELS :read_label TO user",
+        ],
+        [
+            "GRANT CREATE_DELETE ON LABELS :read_label TO user;",
+            "DENY READ ON LABELS :read_label TO user",
+        ],
+        [
+            "GRANT CREATE_DELETE ON LABELS * TO user;",
+            "DENY READ ON LABELS :read_label TO user",
+        ],
+    ]
+
+    match_queries = [match_query, match_by_id_query]
+
+    for grant_group in grant_groups:
+        reset_permissions(admin_cursor)
+
+        [execute_and_fetch_all(admin_cursor, x) for x in grant_group]
+        for mq in match_queries:
+            results = execute_and_fetch_all(user_cursor, mq)
+            assert len(results) == 0
+
+
+def test_can_not_read_node_with_index():
+    admin_cursor = get_admin_cursor()
+    user_cursor = get_user_cursor()
+
+    grant_groups = [
+        [],
+        ["DENY READ ON LABELS :read_label TO user;"],
+        ["DENY READ ON LABELS * TO user;"],
+        [
+            "GRANT UPDATE ON LABELS :read_label TO user;",
+            "DENY READ ON LABELS :read_label TO user",
+        ],
+        [
+            "GRANT UPDATE ON LABELS * TO user;",
+            "DENY READ ON LABELS :read_label TO user",
+        ],
+        [
+            "GRANT CREATE_DELETE ON LABELS :read_label TO user;",
+            "DENY READ ON LABELS :read_label TO user",
+        ],
+        [
+            "GRANT CREATE_DELETE ON LABELS * TO user;",
+            "DENY READ ON LABELS :read_label TO user",
+        ],
+    ]
+
+    match_queries = [
+        match_by_label_query,
+        match_by_label_property_query,
+        match_by_label_property_range_query,
+        match_by_label_property_value_query,
+    ]
+
+    for grant_group in grant_groups:
+        reset_permissions(admin_cursor)
+        execute_and_fetch_all(admin_cursor, "CREATE INDEX ON :read_label;")
+        execute_and_fetch_all(admin_cursor, "CREATE INDEX ON :read_label(prop);")
+
+        [execute_and_fetch_all(admin_cursor, x) for x in grant_group]
```

Comment:
What are we doing with this list? Use simple for loop, the intent is more clear that way. Here some phantom array is created and then not used